app/scripts/detect-multiple-instances.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Sets up two-way communication between the
+ * mainline version of extension and Flask build
+ * in order to detect & warn if there are two different
+ * versions running simultaneously.
+ */
+
+import extension from 'extensionizer';
+
+// TODO: add IDs from other browsers
+const PROD_BUILD_ID = 'nkbihfbeogaeaoehlefnkodbefgpgknn';
+const FLASK_BUILD_ID = 'ljfoeinjpaedjfecbmggjgodbgkmjkjk';
+
+const MESSAGE_TEXT = 'isRunning';
+
+/**
+ * Handles the ping message sent from other extension.
+ */
+export const onMessageReceived = (message, sender, sendResponse) => {
+  if (message === MESSAGE_TEXT) {
+    sendResponse({
+      type: 'success',
+      isRunning: true,
+    });
+    return true;
+  }
+  return false;
+};
+
+/**
+ * Sends the ping message sent to other extension to detect whether it's active or not.
+ * Displays console warning if it's active.
+ */
+export const onExtensionConnect = () => {
```

Comment:
Shouldn't this just run when the application loads and not `onConnect`? 

According to the docs it's trigged by:
> Fired when a connection is made from either an extension process or a content script (by [runtime.connect](https://developer.chrome.com/docs/extensions/reference/runtime/#method-connect)).

So with this current setup I assume yours is running when we connect to other parts of the MM app? 🤔

Reply:
I was thinking of checking for multiple instances every time extension is opened/expanded (after losing focus) – so this is used for that

Reply:
Removed running a check on connection. Now it's startup-only. Also not responding to a ping message (but warning will be now printed only in one console instead of both).